caom2/caom2/tests/test_obs_reader_writer.py

```diff
@@ -449,15 +476,165 @@ def compare_planes(self, expected, actual, version):
                                     actual_plane.provenance)
             self.compare_metrics(expected_plane.metrics, actual_plane.metrics)
             
-            self.compare_time(expected_plane.time, actual_plane.time)
-            
             if version == 21:
                 self.compare_quality(expected_plane.quality,
                                      actual_plane.quality)
+            
+            if version >= 22:
+                self.compare_position(expected_plane.position, actual_plane.position)
+                self.compare_energy(expected_plane.energy, actual_plane.energy)
+                print('comparing time')
+                self.compare_time(expected_plane.time, actual_plane.time)
+                print('compared time')
+                self.compare_polarization(expected_plane.polarization, actual_plane.polarization)
 
             self.compare_artifacts(expected_plane.artifacts,
                                    actual_plane.artifacts, version)
-
+            
+    def compare_none_or_equal(self, expected, actual, field):
+        if expected is None:
+            self.assertIsNone(actual, field)
+        else:
+            self.assertEqual(expected, actual, field)
+            
+    def compare_position(self, expected, actual):
+        if expected is None:
+            self.assertIsNone(actual, "position")
+        else:
+            self.compare_shape(expected.bounds, actual.bounds)
+            self.compare_dimension2d(expected.dimension, actual.dimension)
+            if expected.resolution is None:
```

Comment:
And another overall comment. Variable local to functions should not be prefixed with _. That prefix it's a convention to tell the user that it's a private field/method and it shouldn't be worked with directly. 

You probably used that notation to match the style of the original code that was there.